templates/nextjs/src/pages/index.tsx

```diff
@@ -1,91 +1,46 @@
+import readMessage from '@/utils/readMessage';
+import writeMessage from '@/utils/writeMessage';
 import Head from 'next/head';
 import React, { useState } from 'react';
 import { SpinnerRoundFilled } from 'spinners-react';
-import { Pact, signWithChainweaver } from '@kadena/client';
 import styles from '../styles/main.module.css';
 
-const NETWORK_ID = process.env.NEXT_PUBLIC_KADENA_NETWORK_ID;
-const CHAIN_ID = process.env.NEXT_PUBLIC_KADENA_CHAIN_ID;
-const HOST = process.env.NEXT_PUBLIC_KADENA_HOST;
-const API_HOST = `https://${HOST}/chainweb/0.0/${NETWORK_ID}/chain/${CHAIN_ID}/pact`;
-
-const accountKey = (account: string): string => account.split(':')[1];
-
 const Home: React.FC = (): JSX.Element => {
   const [account, setAccount] = useState<string>('');
   const [messageToWrite, setMessageToWrite] = useState<string>('');
   const [messageFromChain, setMessageFromChain] = useState<string>('');
   const [writeInProgress, setWriteInProgress] = useState<boolean>(false);
 
-  const handleAccountInputChange = (
-    event: React.ChangeEvent<HTMLInputElement>,
-  ): void => {
+  const handleAccountInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
     setAccount(event.target.value);
   };
 
-  const handleWriteMessageInputChange = (
-    event: React.ChangeEvent<HTMLTextAreaElement>,
-  ): void => {
+  const handleWriteMessageInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>): void => {
     setMessageToWrite(event.target.value);
   };
 
-  const writeMessage = async (): Promise<void> => {
-    try {
-      const transactionBuilder = Pact.modules['free.cka-message-store']
-        ['write-message'](account, messageToWrite)
-        .addCap('coin.GAS', accountKey(account))
-        .addCap(
-          'free.cka-message-store.ACCOUNT-OWNER',
-          accountKey(account),
-          account,
-        )
-        .setMeta(
-          {
-            ttl: 28000,
-            gasLimit: 100000,
-            chainId: CHAIN_ID,
-            gasPrice: 0.000001,
-            sender: account,
-          },
-          NETWORK_ID,
-        );
-
-      setWriteInProgress(true);
-
-      await signWithChainweaver(transactionBuilder);
-
-      console.log(`Sending transaction: ${transactionBuilder.code}`);
-      const response = await transactionBuilder.send(API_HOST);
-
-      console.log('Send response: ', response);
-      const requestKey = response.requestKeys[0];
-
-      const pollResult = await transactionBuilder.pollUntil(API_HOST, {
-        onPoll: async (transaction, pollRequest): Promise<void> => {
-          console.log(`Polling ${requestKey}.\nStatus: ${transaction.status}`);
-          console.log(await pollRequest);
-        },
+  function handleWriteMessageClick() {
+    setWriteInProgress(true);
+    writeMessage({ account, messageToWrite })
+      .then(() => {
+        setWriteInProgress(false);
```

Comment:
could be moved to finally unless it absolutely needs to be called before `setMessageToWrite`. btw, why `then().catch()` when we can `async/await`?

Reply:
not have any specific reason for promise catch. I think when i was following migration of kadena client examples are catch so just followed along. thanks for bringing it up refactored to async/await